### ISSUE TITLE
Add mercator profile for output tiles

### ISSUE THREAD

**Issue by Gigas002** [Gigas002/GTiff2Tiles#6]
**Why**
To implement more functionality of original [gdal2tiles.py](https://github.com/OSGeo/gdal/blob/master/gdal/swig/python/scripts/gdal2tiles.py), that would be good to create an option to make mercator tiles too.

**Where to start / alternatives**
The [gdal2tiles.py](https://github.com/OSGeo/gdal/blob/master/gdal/swig/python/scripts/gdal2tiles.py) is a good base point, some maths can be used from it and rewritten on C#.

**Comment by Gigas002** [comment]
Approximate plan looks like this:
- [ ] Open inout geotiff;
- [ ] Check projection and stuff;
- [ ] Convert to mercator, if needed;
- [ ] Get mercator coordinates from file metadata;
- [ ] Calculate mercator tile numbers dictionary for each zoom;
- [ ] Generate tiles;

**Comment by carnegiea** [comment]
Any help I can provide in this matter? It seems like my previous issue was due to mercator / geodesic tile numbering as well, therefore i got confused even at openlayer.

**Comment by Gigas002** [comment]
**TLDR**: I'm not yet working on that feature.

Long explanation:
Well, I supposes 1-4 paragraphs are pretty easy to implement through gdal library in-code. The difficulties are starting with maths, when calculating tile numbers from mercator coordinates and calculating zones for tiles to read/write.
I haven't yet looked on all that stuff, but I'm pretty sure there's enought info about how-to-do-that in original [gdal2tiles.py](https://github.com/OSGeo/gdal/blob/master/gdal/swig/python/scripts/gdal2tiles.py) script.
The point is also not to change too much in **GTiff2Tiles.Core** existing code, so that'll take some time to implement. I should also change some names because of that (for example `RepairTifOptions` enum is needed only to convert tif to `geodetic`, but it will be kind of confusing to use that property to "reapair" `mercator` tif).

**Comment by Gigas002** [comment]
Logics in `GTiff2Tiles.Core` for this feature complete and will be availibale in **GTiff2Tiles nuget package** starting from *v2.0.0-ci436 or higher*.
It'll be added to `GTiff2Tiles.Console` and `GTiff2Tiles.GUI` in next major stable release (*2.0.0*).

**Comment by Gigas002** [comment]
As I said above, work in Core for this feature complete and new options will appear in new stable release in both Console and GUI apps. 
To avoid confusion, I'm closing this issue.
